ibverbs/src/lib.rs

```diff
@@ -1226,15 +1384,173 @@ impl<'res> QueuePair<'res> {
+            __bindgen_anon_2: Default::default(),
+        };
+        let mut bad_wr: *mut ffi::ibv_send_wr = ptr::null::<ffi::ibv_send_wr>() as *mut _;
+
+        let ctx = (&*self.qp).context;
+        let ops = &mut (&mut *ctx).ops;
+        let errno =
+            ops.post_send.as_mut().unwrap()(self.qp, &mut wr as *mut _, &mut bad_wr as *mut _);
+        if errno != 0 {
+            Err(io::Error::new(
+                io::ErrorKind::Other,
+                format!("ERROR: Faile to write. errno={}", errno),
+            ))
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Write a single value from the top of the memory region
+    #[inline]
+    pub unsafe fn post_read_buf<T>(
+        &self,
+        mr: &MemoryRegion<T>,
+        n: usize,
+        addr: u64,
+        key: u32,
+        wr_id: u64,
+        sig: bool,
+    ) -> io::Result<()> {
+        assert!(
+            mr.size >= n,
+            "ERROR: cannot read {} values to a MR of size {}, MR is too small",
+            n,
+            mr.size
+        );
+
+        let mut sge = ffi::ibv_sge {
+            addr: mr.as_ptr() as u64,
+            length: (mem::size_of::<T>() * n) as u32,
+            lkey: (&*mr.mr).lkey,
+        };
+        let flg = if sig {
+            ffi::ibv_send_flags::IBV_SEND_SIGNALED.0
+        } else {
+            0
+        };
+
+        let mut wr = ffi::ibv_send_wr {
+            wr_id: wr_id,
+            next: ptr::null::<ffi::ibv_send_wr>() as *mut _,
+            sg_list: &mut sge as *mut _,
+            num_sge: 1,
+            opcode: ffi::ibv_wr_opcode::IBV_WR_RDMA_READ,
+            send_flags: flg,
+            wr: ffi::ibv_send_wr__bindgen_ty_2 {
+                rdma: ffi::ibv_send_wr__bindgen_ty_2__bindgen_ty_1 {
+                    remote_addr: addr,
+                    rkey: key,
+                },
+            },
+            qp_type: Default::default(),
+            __bindgen_anon_1: Default::default(),
+            __bindgen_anon_2: Default::default(),
+        };
+        let mut bad_wr: *mut ffi::ibv_send_wr = ptr::null::<ffi::ibv_send_wr>() as *mut _;
+
+        let ctx = (&*self.qp).context;
+        let ops = &mut (&mut *ctx).ops;
+        let errno =
+            ops.post_send.as_mut().unwrap()(self.qp, &mut wr as *mut _, &mut bad_wr as *mut _);
+        if errno != 0 {
+            Err(io::Error::new(
+                io::ErrorKind::Other,
+                format!("ERROR: Faile to read. errno={}", errno),
+            ))
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Write a single value from the top of the memory region
+    #[inline]
+    pub unsafe fn post_write_single<T>(
+        &self,
+        mr: &MemoryRegion<T>,
+        addr: u64,
+        key: u32,
+        wr_id: u64,
+        sig: bool,
+    ) -> io::Result<()> {
+        self.post_write_buf(mr, 1, addr, key, wr_id, sig)
+    }
+
+    /// Write a single value from the top of the memory region
```

Comment:
This could use some more documentation. Especially with regards to why it is `unsafe`.